components/org.wso2.carbon.identity.api.server.application.management/org.wso2.carbon.identity.api.server.application.management.v1/src/main/java/org/wso2/carbon/identity/api/server/application/management/v1/core/functions/application/inbound/saml/SAMLInboundFunctions.java

```diff
@@ -171,6 +173,50 @@ public static InboundAuthenticationRequestConfig createSAMLInbound(SAML2Configur
         return samlInbound;
     }
 
+    public static InboundAuthenticationRequestConfig createSAMLInbound(
+            ServiceProvider serviceProvider, SAML2Configuration saml2Configuration) {
+
+        SAML2ServiceProvider samlManualConfiguration = saml2Configuration.getManualConfiguration();
+
+        SAMLSSOServiceProviderDTO samlssoServiceProviderDTO;
+        if (saml2Configuration.getMetadataFile() != null) {
+            samlssoServiceProviderDTO = createSAMLSpWithMetadataFile(saml2Configuration.getMetadataFile());
+        } else if (saml2Configuration.getMetadataURL() != null) {
+            samlssoServiceProviderDTO = createSAMLSpWithMetadataUrl(saml2Configuration.getMetadataURL());
+        } else if (samlManualConfiguration != null) {
+            samlssoServiceProviderDTO = createSAMLSpWithManualConfiguration(samlManualConfiguration);
+        } else {
+            throw Utils.buildBadRequestError("Invalid SAML2 Configuration. One of metadataFile, metaDataUrl or " +
+                    "serviceProvider manual configuration needs to be present.");
+        }
```

Comment:
Would it make sense to move this segment to a private method to keep the method clean and clear?

Reply:
Fixed